### ISSUE TITLE
logger is logging reponse bodies for request bodies

### ISSUE THREAD

**Issue by bughit** [oauth-xx/oauth2#214]
by the time logger is called, env[:body] already contains the response body

**Comment by bughit** [comment]
```ruby
      #connection.response :logger, ::Logger.new($stdout) if ENV['OAUTH_DEBUG'] == 'true'
      connection.builder.insert_before 0, Faraday::Response::Logger, ::Logger.new($stdout) if ENV['OAUTH_DEBUG'] == 'true'
```
inserting the logger earlier seems to work

**Comment by bughit** [comment]
the logger needs to be just before the default adapter of the default stack, in order to log the encoded request body

```ruby
connection.builder.insert_before connection.builder.handlers.length - 1, Faraday::Response::Logger, ::Logger.new($stdout) if ENV['OAUTH_DEBUG'] == 'true'
```

```ruby
        # default stack, if nothing else is configured
        self.request :url_encoded
        self.adapter Faraday.default_adapter
```

**Comment by xtagon** [comment]
@bughit Did this fix your issue? Would you be willing to do a PR if so?

**Comment by pboling** [comment]
Implemented in #366
